### ISSUE TITLE
SessionRefresh middleware is not usable with other authentication backends

### ISSUE THREAD

**Issue by gavinwahl** [mozilla/mozilla-django-oidc#248]
My project allows both oidc authentication and regular django username/password authentication. Everything works great except for SessionRefresh middleware, which is redirecting my normal users through the oidc process. I think that the only thing need to fix this is to not try to refresh the token if oidc_id_token_expiration is not in the session. I tried that and it works, but there is a test case specifically forbidding this behavior. Why is that? oidc_id_token_expiration will be in the session iff the user is an oidc user, so I don't see why we need to have a case for when it's missing.

**Comment by gavinwahl** [comment]
I would suggest this patch to fix the issue:

```patch

diff --git a/mozilla_django_oidc/middleware.py b/mozilla_django_oidc/middleware.py
index e01664c..b84c397 100644
--- a/mozilla_django_oidc/middleware.py
+++ b/mozilla_django_oidc/middleware.py
@@ -85,7 +85,11 @@ class SessionRefresh(MiddlewareMixin):
             LOGGER.debug('request is not refreshable')
             return
 
-        expiration = request.session.get('oidc_id_token_expiration', 0)
+        try:
+            expiration = request.session['oidc_id_token_expiration']
+        except KeyError:
+            # User is not authenticated with OpenID.
+            return
         now = time.time()
         if expiration > now:
             # The id_token is still valid, so we don't have to do anything.
```

**Comment by johngian** [comment]
Fixed after #259
